src/server/database/types/GenericMeetingStage.ts

```diff
@@ -1,13 +1,48 @@
 import shortid from 'shortid'
+import ms from 'ms'
+
+const MAX_SYNC_STAGE_DURATION = ms('1h')
+
+// https://stackoverflow.com/a/20811670/3155110
+const filterOutliers = (someArray: number[]) => {
+  const values = someArray.concat()
+  values.sort(function (a, b) {
+    return a - b
+  })
+  const q1 = values[Math.floor(values.length / 4)]
+  const q3 = values[Math.ceil(values.length * (3 / 4))]
+  const iqr = q3 - q1
+
+  const maxValue = q3 + iqr * 1.5
+  const minValue = q1 - iqr * 1.5
+  return values.filter((x) => x <= maxValue && x >= minValue)
+}
 
 export default class GenericMeetingStage {
   id = shortid.generate()
-  meetingId
+  isAsync: boolean | undefined | null
   isComplete = false
   isNavigable = false
   isNavigableByFacilitator = false
   startAt: Date | undefined = undefined
   endAt: Date | undefined = undefined
+  scheduledEndTime: Date | undefined | null
+  suggestedEndTime: Date | undefined
+  suggestedTimeLimit: number | undefined
   viewCount = 0
-  constructor (public phaseType: string) {}
+
+  constructor (public phaseType: string, durations: number[] | undefined) {
```

Comment:
`phaseType` doesn't appear to be used here. Are you thinking you'd use it in the future?